src/crypto/abe/hybrid_crypto/interfaces/decryption.ts

```diff
@@ -0,0 +1,66 @@
+import { ClearTextHeader } from "../cleartext_header"
+
+
+
+export abstract class HybridDecryption {
+  private _asymmetricDecryptionKey: Uint8Array
+
+  public set asymmetricDecryptionKey(value: Uint8Array) {
+    this._asymmetricDecryptionKey = value
+  }
+  public get asymmetricDecryptionKey(): Uint8Array {
+    return this._asymmetricDecryptionKey
+  }
+
+  constructor(asymmetricDecryptionKey: Uint8Array) {
+    this._asymmetricDecryptionKey = asymmetricDecryptionKey
+  }
+
+  public abstract renewKey(userDecryptionKey: Uint8Array): void
+
+  public abstract destroyInstance(): void
+
+  /**
+   *
+   * @param asymmetricHeader asymmetric encrypted data
+   */
+  public abstract decryptHybridHeader(asymmetricHeader: Uint8Array): ClearTextHeader
+
+  /**
+   * Decrypts a hybrid block
+   *
+   * @param symmetricKey symmetric key
+   * @param encryptedBytes encrypted data
+   * @param uid uid used as additional data
+   * @param blockNumber
+   * @returns the cleartext if everything succeeded
+   */
+  public abstract decryptHybridBlock(symmetricKey: Uint8Array, encryptedBytes: Uint8Array, uid: Uint8Array | undefined, blockNumber: number | undefined): Uint8Array
```

Comment:
I think it's better to use `null` than `undefined` 

> The value null indicates that you know that the field does not have a value. It is an intentional absence of value.

Reply:
I am not sure for this one: `wasm-pack build` generates JS-bindings as `Uint8Array | undefined` when Rust arguments are optional.